morpheus/models/dfencoder/ae_feature_selector.py

```diff
@@ -0,0 +1,538 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+import numpy as np
+import pandas as pd
+import torch
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_selection import VarianceThreshold
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+from tqdm.notebook import tqdm
+
+from morpheus.models.dfencoder.autoencoder import AutoEncoder
+
+LOG = logging.getLogger('AE Feature Selector')
+
+
+class AutoencoderFeatureSelector:
+    """
+    A class to select features using an autoencoder, handling categorical and numerical data.
+    Supports handling ambiguities in data types and ensures selected feature count constraints.
+
+    Attributes:
+        input_json (dict): List of dictionary objects to normalize into a dataframe
+        id_column (str) : Column name that contains ID for morpheus AE pipeline. Default None.
+        timestamp_column (str) : Column name that contains the
+            timestamp for morpheus AE pipeline. Default None.
+        encoding_dim (int): Dimension of the encoding layer, defaults to half of input
+            dimensions if not set.
+        batch_size (int): Batch size for training the autoencoder.
+        variance_threshold (float) : Minimum variance a column must contain
+            to remain in consideration. Default 0.
+        null_threshold (float): Maximum proportion of null values a column can contain. Default 0.3.
+        cardinality_threshold_high (float): Maximum proportion
+            of cardinality to length of data allowable. Default 0.99.
+        cardinality_threshold_low_n (int): Minimum cardinalty for a feature to be considered numerical
+            during type infernce. Default 10.
+        categorical_features (list[str]): List of features in the data to be considered categorical. Default [].
+        numeric_features (list[str]): List of features in the data to be considered numeric. Default [].
+        ablation_epochs (int): Number of epochs to train the autoencoder.
+        device (str): Device to run the model on, defaults to 'cuda' if available.
+
+    Methods:
+        train_autoencoder(data_loader): Trains the autoencoder using the provided data loader.
+        calculate_loss(model, data_loader): Calculates reconstruction loss using the trained model.
+        preprocess_data(): Preprocesses the DataFrame to handle numerical and categorical data appropriately.
+        remove_low_variance(processed_data): Removes features with low variance.
+        remove_high_correlation(data): Removes highly correlated features based on a threshold.
+        feature_importance_evaluation(processed_data): Evaluates feature importance using the autoencoder.
+        select_features(k_min, k_max): Selects features based on importance, adhering to min/max constraints.
```

Comment:
This docstring is in the wrong format. Convert it to follow the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) specification